utilities/include/vec3.hpp

```diff
@@ -1,212 +1,66 @@
-            return y;
-        }
+    public:
+        constexpr Vec3(Utype x,Utype y,Utype z) : Vec4{x,y,z,0} {}
+        constexpr explicit Vec3(xsimd::batch<Utype,UArch> x) : Vec4{x} {}
         
-        constexpr float Z() const{
-            return z;
+        // cross-product
+        constexpr Vec3 cross(const Vec3& v2) const{
+
+            xsimd::batch<Utype,UArch> B0 = vec2batch(*this); 
+            xsimd::batch<Utype,UArch> B1= vec2batch(v2);
+            
+            //create a appropriate mask for cross product.
+            auto shuffler =  xsimd::make_batch_constant<typename as_index<xsimd::batch<float, xsimd::sse4_2>>::type, indices>();
+            
+            xsimd::batch<Utype,UArch> temp0 = xsimd::swizzle(B1,shuffler);
+            xsimd::batch<Utype,UArch> temp1 = xsimd::swizzle(B0,shuffler);
+
+            temp0 = temp0*B0;
+            temp1 = temp1*B1;
+
+            return batch2vec3(xsimd::swizzle(temp0-temp1,shuffler));
         }
 
-        //vector addition
-        friend constexpr vec3 operator+(const vec3& v1, const vec3& v2);
-
-        //vector subtraction
-        friend constexpr vec3 operator-(const vec3& v1, const vec3& v2);
-
-        //scalar post-multiplication
-        friend constexpr vec3 operator*(const vec3& v1, const float& s);
-
-        //scalar pre-multiplication
-        friend constexpr vec3 operator*(const float& s, const vec3& v1);
-
-        //scalar post-division
-        friend constexpr vec3 operator/(const vec3& v1, const float& s);
-
-        //addition+assignemnt
-        friend constexpr vec3& operator+=(vec3& v1, const vec3& v2);
-        
-        //subtraction+assignemnt
-        friend constexpr vec3& operator-=(vec3& v1, const vec3& v2);
-    
-        //scalar multiplication+assignment
-        friend constexpr vec3& operator*=(vec3& v1, const float& s);
-        
-        //scalar division+assignment
-        friend constexpr vec3& operator/=(vec3& v1, const float& s);
-
-        //negation operator 
-        friend constexpr vec3 operator-(const vec3& v1);
-
-        //dot product
-        friend constexpr float dot(const vec3& v1, const vec3& v2);
-
-        //cross product
-        friend constexpr vec3 cross(const vec3& v1, const vec3& v2);
-
-        //angle between vectors, in radians
-        #ifndef __GNUC__ 
-
-        friend float angle(const vec3& v1, const vec3& v2);
-
-        #endif
-        
-        #ifdef __GNUC__
-
-        friend constexpr float angle(const vec3& v1, const vec3& v2);
-
-        #endif
-
-        //projection of h on s
-        friend constexpr vec3 proj(const vec3& v1, const vec3& v2);
+    private:
+        static constexpr Vec3 batch2vec3(xsimd::batch<Utype,UArch> x){
+            Vec3 v{x};
+            return v;
+        }
 
-        //perpendicular projection
-        friend constexpr vec3 perp(const vec3& v1, const vec3& v2);
 
-    
-        friend std::ostream& operator<<(std::ostream& out, const vec3& vect);
 
```

Comment:
Same goes here. Too many blank lines here.

Reply:
you are referring to old vec3 file